### PR TITLE
refactor: update nodejs-common and google-auth-library

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -404,7 +404,7 @@ function Datastore(options) {
     this.options.sslCreds = grpc.credentials.createInsecure();
   }
 
-  this.auth = googleAuth(this.options);
+  this.auth = new GoogleAuth(this.options);
```
Comment (stephenplusplus) [inline src/index.js]: There is a use of this in `src/request.js`, that I don't think is compatible with `GoogleAuth`: https://github.com/googleapis/nodejs-datastore/blob/8ef7e74a540a87b49a9bc6ce46d7947e071403e9/src/request.js#L1133

Comment (JustinBeckwith) [reply src/index.js]: I think it's ok.  `getDefaultProjectId` will accept a callback:
https://github.com/google/google-auth-library-nodejs/blob/master/src/auth/googleauth.ts#L137

And it's aliased to `getProjectId`:
https://github.com/google/google-auth-library-nodejs/blob/master/src/auth/googleauth.ts#L91

Comment (JustinBeckwith) [reply src/index.js]: Forgot to add - all the system tests pass :)

Comment (stephenplusplus) [reply src/index.js]: Nice! I tested in the console, but must have done something wrong.

**Pull request merged**